Automattic/jetpack
# Connection Banner: dismiss button accessibility

Fixes https://github.com/Automattic/jetpack/issues/35480

## Proposed changes:

We need a better description for Dismiss button on Connection Banner
<img width="687" alt="Screenshot 2024-02-14 at 11 43 42 AM" src="https://github.com/Automattic/jetpack/assets/60262784/97f8a433-acd9-4e18-808d-21f0a2a17962">


### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Create a new JN site using this branch
* Navigate to My Jeetpack page
* Check `Welcome to Jetpack` banner with screen reader. The Close button should read: 'Don’t show the welcome message again'
